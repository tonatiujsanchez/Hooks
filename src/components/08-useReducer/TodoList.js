import TodoListItem from "./TodoListItem"

import PropTypes from 'prop-types'


const TodoList = ({ todos, handleToggle, handleDelete  }) => {
    return (
        <ul className='list-group list-group-flush'>
            {todos.map((todo, idx) => (
                <TodoListItem 
                    key={ todo.id } 
                    todo={ todo }
                    idx={ idx } 
                    handleToggle={ handleToggle } 
                    handleDelete={ handleDelete } />
            ))}
        </ul>
    )
}

TodoList.propTypes = {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
}

export default TodoList