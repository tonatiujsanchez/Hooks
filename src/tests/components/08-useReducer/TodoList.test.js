import { shallow } from "enzyme"
import TodoList from "../../../components/08-useReducer/TodoList"
import { demoTodos } from "../../fixtures/demoTodos"



describe('Pruebas en <TodoList />', () => { 

    const handleToggle = jest.fn()
    const handleDelete = jest.fn()

    const wrapper = shallow( 
        <TodoList
            todos={ demoTodos }
            handleToggle = { handleToggle }
            handleDelete = { handleDelete }
        />
    )

    test('Debe de mostrarse correctamente', () => { 
        
        expect( wrapper ).toMatchSnapshot()

     })

     test('Debe tene 2 <TodoListItem />', () => { 



        const todos = wrapper.find('TodoListItem')
        
        expect( todos.length ).toBe( demoTodos.length )

        const hDelete = todos.at(0).prop('handleDelete')

        expect(hDelete).toEqual( expect.any(Function) )
        expect( typeof hDelete ).toBe('function')

      })


 })