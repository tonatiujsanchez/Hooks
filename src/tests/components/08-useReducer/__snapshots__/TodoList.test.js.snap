// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    idx={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    idx={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Next",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
